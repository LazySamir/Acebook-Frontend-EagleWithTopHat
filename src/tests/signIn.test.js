import React from "react";
import SignIn from '../components/signIn';
import {shallow, mount, render} from 'enzyme'

describe('signIn', () => {
  const wrapper = shallow(<SignIn />)

  describe('email', () => {
    const emailState = wrapper.state().email
    it('starts empty', () => {
      expect(emailState).toEqual("")
    })

    it('recognises keyboard input', () => {
      wrapper.find('[name="email"]').simulate('click')

      expect(wrapper).toMatchSnapshot();
    })

    it('changes state on keyboard input', () => {
      wrapper.find('[name="email"]').simulate('change', { target: {name:'email', value: 'fakeemail@fakeemail.com' }})

      expect(wrapper.state('email')).toEqual('fakeemail@fakeemail.com')
    })
  })

  describe('password', () => {
    const passwordState = wrapper.state().password
    it('starts empty', () => {
      expect(passwordState).toEqual("")
    })

    it('recognises keyboard input', () => {
      wrapper.find('[name="password"]').simulate('click')

      expect(wrapper).toMatchSnapshot();
    })

    it('changes state on keyboard input', () => {
      wrapper.find('[name="password"]').simulate('change', { target: {name:'password', value: 'fakepassword' }})

      expect(wrapper.state('password')).toEqual('fakepassword')
    })
  })

  describe('submit', () => {

    it('returns all post objects as json', () => {

      // expect(response.content_type).toEqual("application/json")
    })
  })
})
