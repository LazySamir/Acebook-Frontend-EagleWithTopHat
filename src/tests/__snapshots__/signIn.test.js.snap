// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signIn email recognises keyboard input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <label>
          Email:
          <input
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />
        </label>
        <br />
        <label>
          Password:
          <input
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </label>
        <br />
        <input
          type="submit"
          value="Submit"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Email:
            <input
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
          </label>,
          <br />,
          <label>
            Password:
            <input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </label>,
          <br />,
          <input
            type="submit"
            value="Submit"
          />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Email:",
              <input
                name="email"
                onChange={[Function]}
                type="email"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Email:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Password:",
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Password:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
            "value": "Submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <label>
            Email:
            <input
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
          </label>
          <br />
          <label>
            Password:
            <input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </label>
          <br />
          <input
            type="submit"
            value="Submit"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Email:
              <input
                name="email"
                onChange={[Function]}
                type="email"
                value=""
              />
            </label>,
            <br />,
            <label>
              Password:
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </label>,
            <br />,
            <input
              type="submit"
              value="Submit"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Email:",
                <input
                  name="email"
                  onChange={[Function]}
                  type="email"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Email:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Password:",
                <input
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Password:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`signIn password recognises keyboard input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <label>
          Email:
          <input
            name="email"
            onChange={[Function]}
            type="email"
            value="fakeemail@fakeemail.com"
          />
        </label>
        <br />
        <label>
          Password:
          <input
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </label>
        <br />
        <input
          type="submit"
          value="Submit"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Email:
            <input
              name="email"
              onChange={[Function]}
              type="email"
              value="fakeemail@fakeemail.com"
            />
          </label>,
          <br />,
          <label>
            Password:
            <input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </label>,
          <br />,
          <input
            type="submit"
            value="Submit"
          />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Email:",
              <input
                name="email"
                onChange={[Function]}
                type="email"
                value="fakeemail@fakeemail.com"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Email:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "type": "email",
                "value": "fakeemail@fakeemail.com",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Password:",
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Password:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
            "value": "Submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <label>
            Email:
            <input
              name="email"
              onChange={[Function]}
              type="email"
              value="fakeemail@fakeemail.com"
            />
          </label>
          <br />
          <label>
            Password:
            <input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </label>
          <br />
          <input
            type="submit"
            value="Submit"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Email:
              <input
                name="email"
                onChange={[Function]}
                type="email"
                value="fakeemail@fakeemail.com"
              />
            </label>,
            <br />,
            <label>
              Password:
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </label>,
            <br />,
            <input
              type="submit"
              value="Submit"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Email:",
                <input
                  name="email"
                  onChange={[Function]}
                  type="email"
                  value="fakeemail@fakeemail.com"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Email:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "type": "email",
                  "value": "fakeemail@fakeemail.com",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Password:",
                <input
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Password:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
